include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -389,7 +389,9 @@ struct __gen_set_mask
 
         std::size_t __nb = __set_b.size();
 
-        const auto& __val_a_proj = std::invoke(__proj1, __set_a[__id]);
+        // This local variable extends the lifetime of operator[] return in case of temporary object
```

Comment:
```suggestion
        // This reference extends the lifetime of a temporary object returned by operator[]
        // so that it can be safely used with identity projections
```

Reply:
fixed